src/components/dialogs/script-dialog.js

```diff
@@ -74,61 +77,91 @@ const ScriptContingencyDialog = ({ listId, open, onClose, onError, title }) => {
     };
 
     const handleCancel = () => {
-        setAceEditorContent(currentScriptContingency.script);
+        setAceEditorContent(currentScript.script);
         setBtnSaveListDisabled(true);
         onClose();
     };
 
     const handleClick = () => {
-        let newScriptContingencyList = {
-            id: listId,
-            name: name,
-            description: description,
-            script: aceEditorContent,
-        };
-        saveScriptContingencyList(newScriptContingencyList)
-            .then((response) => {})
-            .catch((error) => {
-                onError(error.message);
-            });
+        let newScript;
+        if (type === elementType.SCRIPT_CONTINGENCY_LIST) {
+            newScript = {
+                id: id,
+                name: name,
+                description: description,
+                script: aceEditorContent,
+            };
+            saveScriptContingencyList(newScript)
+                .then((response) => {})
+                .catch((error) => {
+                    onError(error.message);
+                });
+        } else {
+            newScript = {
+                id: id,
+                name: name,
+                description: description,
+                script: aceEditorContent,
+                type: elementType.SCRIPT,
+            };
+            saveFilter(newScript)
+                .then((unused) => {})
+                .catch((error) => {
+                    onError(error.message);
+                });
+        }
         onClose();
-        setCurrentScriptContingency(newScriptContingencyList);
+        setCurrentScript(newScript);
     };
 
     const onChangeAceEditor = (newScript) => {
         setAceEditorContent(newScript);
-        if (
-            currentScriptContingency !== null &&
-            newScript !== currentScriptContingency.script
-        ) {
+        if (currentScript !== null && newScript !== currentScript.script) {
             setBtnSaveListDisabled(false);
         } else {
             setBtnSaveListDisabled(true);
         }
     };
 
-    const getCurrentContingencyList = useCallback(
+    const getCurrentScript = useCallback(
         (currentItemId) => {
-            getContingencyList(ScriptTypes.SCRIPT, currentItemId)
-                .then((data) => {
-                    if (data) {
-                        setCurrentScriptContingency(data);
-                        setAceEditorContent(data.script);
-                        setName(data.name);
-                        setDescription(data.description);
-                    }
-                })
-                .catch((error) => {
-                    onError(error.message);
-                });
+            if (type === elementType.SCRIPT_CONTINGENCY_LIST) {
```

Comment:
Not sure if its here that we have to manage a type and conditionally call APIs by type to get or save scripts... maybe we should manage this outside this component which is fed with a scriptdata, controlled prop and a onClick prop for save.